velox/functions/prestosql/window/Rank.cpp

```diff
@@ -41,8 +44,25 @@ class RankFunction : public exec::WindowFunction {
       vector_size_t resultOffset,
       const VectorPtr& result) override {
     int numRows = peerGroupStarts->size() / sizeof(vector_size_t);
-    auto* rawValues = result->asFlatVector<int64_t>()->mutableRawValues();
     auto* rawPeerStarts = peerGroupStarts->as<vector_size_t>();
+
+    void* rawValues = nullptr;
+    switch (R) {
+      case velox::exec::RankType::kRank:
+      case velox::exec::RankType::kDenseRank: {
+        auto* resultRawValues =
```

Comment:
Why did you create this intermediate variable for resultRawValues ? Can we simply say 

rawValues = result->asFlatVector<int64_t>....

Reply:
I thought it made the code more readable, will revert it, thanks!